### PR TITLE
8269074: (fs) Files.copy fails to copy from /proc on some linux kernel versions

### PR DESCRIPTION
**Pull request by bplb**
On Linux, if while executing `java.nio.file.Files.copy(Path,Path,CopyOption...)` the native function `sendfile(2)` fails with `EINVAL`, fall back to transferring via user-space buffers.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269074](https://bugs.openjdk.java.net/browse/JDK-8269074): (fs) Files.copy fails to copy from /proc on some linux kernel versions

### Reviewers
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**) ⚠️ Review applies to 4886f787606de6ed40cc42edd0ff93d02f55fa85
 * [Vyom Tewari](https://openjdk.java.net/census#vtewari) (@vyommani - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk17 pull/128/head:pull/128` \
`$ git checkout pull/128`

Update a local copy of the PR: \
`$ git checkout pull/128` \
`$ git pull https://git.openjdk.java.net/jdk17 pull/128/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 128`

View PR using the GUI difftool: \
`$ git pr show -t 128`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk17/pull/128.diff">https://git.openjdk.java.net/jdk17/pull/128.diff</a>

</details>

### CODE REVIEW

**Review by AlanBateman — APPROVED** [review]

File: src/java.base/unix/native/libnio/fs/UnixCopyFile.c [hunk 1]
```diff
@@ -87,7 +120,12 @@ Java_sun_nio_fs_UnixCopyFile_transfer
     do {
         RESTARTABLE(sendfile64(dst, src, NULL, count), bytes_sent);
         if (bytes_sent == -1) {
-            throwUnixException(env, errno);
+            if (errno == EINVAL) {
```
Comment (vyommani) [inline src/java.base/unix/native/libnio/fs/UnixCopyFile.c:123]: Don't you think we have have to take care of "ENOSYS" as well. Manual pages talks about "ENOSYS" as well.
 
"Applications may wish to fall back to read(2)/write(2) in the case where sendfile() fails with EINVAL or ENOSYS."

Comment (bplb) [reply src/java.base/unix/native/libnio/fs/UnixCopyFile.c:123]: I think that's a good point.

**Review by vyommani — APPROVED** [review]
Looks OK to me.

**Pull request closed without merging**